pxr/imaging/hdMtlx/hdMtlx.cpp

```diff
@@ -585,6 +607,21 @@ HdMtlxCreateMtlxDocumentFromHdMaterialNetworkInterface(
         mxType,
         mxShaderNode);
 
+    if (TfDebug::IsEnabled(HDMTLX_VERSION_UPGRADE)) {
+      writeToXmlFile(
+          mxDoc, mx::FilePath(mxMaterial->getName()+"_before.mtlx"));
```

Comment:
Possibly want to add a printout of which files you're writing to stdout?

Reply:
make sense - added.